### DISCUSSION TITLE
Are Vercel CDN nodes not interconnected with each other?

### DISCUSSION THREAD

**Discussion by abriginets** [vercel/vercel#6976] [Help]
Hello. I've mentioned some strange behavior of Vercel's CDN network. It appears that every single CDN region has it's own cache which is not interconnected with other regions. That's how I figured out - I've made a request from my homestation and then repeated the request with check-host.net (service to test websites availability from different locations) and from near my location CDN responds real quick:
![image](https://user-images.githubusercontent.com/17240507/141188845-18528b2a-bc8d-4631-946a-aff0f783a126.png)
But from any other location it takes incomparably large amount of time to serve:
![image](https://user-images.githubusercontent.com/17240507/141189072-66980294-d218-4a40-9e6c-78b2fd608f93.png)

It really looks like Vercel is deploying my app to the location near the requester, executes the lambda and then serves the response. I doubt it works like that on Pro plan but it seems like the closest node to requester doesn't do the "hey other nodes, do you have this page cached already and maybe I can reuse it?" thing. Should it be the default behavior or am I missing something? AFAIK CloudFlare utilize this strategy.

**Comment by leerob** [comment]
> It appears that every single CDN region has it's own cache which is not interconnected with other regions.

Correct, each location has its own regional cache. When you deploy an application to Vercel, your assets (HTML, JS, CSS) are globally deployed at each region. If you're running a [Serverless Function](https://vercel.com/docs/concepts/functions/serverless-functions), it is deployed to `us-east` by default (Hobby). On Pro, you can pick your region. With [Edge Functions](https://vercel.com/edge), they deploy to every region by default.

[More on caching here.](https://vercel.com/docs/concepts/edge-network/caching)
